modules/core/src/test/java/org/apache/ignite/internal/util/IgniteNameUtilsTest.java

```diff
@@ -165,4 +167,38 @@ private static List<Arguments> parseNameIllegalArguments() {
                 Arguments.of("foo..", "Malformed identifier [identifier=foo.., pos=4]")
         );
     }
+
+    @ParameterizedTest
+    @MethodSource("parseAndNormalizeData")
+    public void parseAndNormalize(String input, String result) {
+        String actual = IgniteNameUtils.parseAndNormalize(input);
+        assertEquals(result, actual);
+    }
+
+    private static Stream<Arguments> parseAndNormalizeData() {
+        return Stream.of(
+                Arguments.of("foo", "FOO"),
+                Arguments.of("Foo", "FOO"),
+                Arguments.of("fOo", "FOO"),
+                Arguments.of("f_oo", "F_OO"),
+                Arguments.of("_foo", "_FOO"),
+                Arguments.of("foo_", "FOO_"),
+                Arguments.of("_foo_", "_FOO_"),
+
+                Arguments.of("foo1", "FOO1"),
+                Arguments.of("foo-1", "\"foo-1\""),
+                Arguments.of("foo_1", "FOO_1"),
+                Arguments.of("1foo", "\"1foo\""),
+                Arguments.of("1_foo", "\"1_foo\""),
+
+                Arguments.of("a bar", "\"a bar\""),
+                Arguments.of("A bar", "\"A bar\""),
+                Arguments.of("A.bar", "\"A.bar\""),
```

Comment:
These are not valid SQL identifiers and must be quoted by a user.

Reply:
totally agree